src/backend/distributed/executor/multi_executor.c

```diff
@@ -182,7 +203,7 @@ CitusExecutorRun(QueryDesc *queryDesc,
 		 * the value because exceptions might cause us to go back a few
 		 * levels at once.
 		 */
```

Comment:
Does the same apply to ExecutorLevel?

Reply:
Currently I increment/decrement in ExecutorStart/End functions. Given your point, maybe I should do it ExecutorRun. I'll update the PR with this after finding a failing test.